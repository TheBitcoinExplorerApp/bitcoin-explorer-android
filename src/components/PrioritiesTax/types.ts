export type FeesType = {
  fastestFee: number;
  halfHourFee: number;
  hourFee: number;
  economyFee: number;
  minimumFee: number;
};
