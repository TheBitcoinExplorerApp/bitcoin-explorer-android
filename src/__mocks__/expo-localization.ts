const getLocales = () => [
  { country: "pt", languageTag: "pt-BR", languageCode: "pt", isRTL: false },
];

export default getLocales;
