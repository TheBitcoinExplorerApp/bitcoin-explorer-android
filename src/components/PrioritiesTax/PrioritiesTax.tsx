// eslint-disable-next-line import/no-extraneous-dependencies
import { View, Text, StyleSheet } from "react-native";
import useAppDataStore from "src/context/DataProvider";
import SmallBoxInfo from "../SmallBoxInfo/SmallBoxInfo";

const styles = StyleSheet.create({
  taxContainerPrioritiesTaxText: {
    width: "100%",
    flexDirection: "row",
    justifyContent: "space-around",
    backgroundColor: "#1d2133",
    padding: 16,
    borderRadius: 8,
  },
  taxPrioritiesTaxText: {
    color: "#FFF",
    fontSize: 12,
  },
});

export default function PrioritiesTax() {
  const { fees, i18n } = useAppDataStore();

  return (
    <>
      <View style={styles.taxContainerPrioritiesTaxText}>
        <Text style={styles.taxPrioritiesTaxText}>
          {i18n.t("low_priority")}
        </Text>
        <Text style={styles.taxPrioritiesTaxText}>
          {i18n.t("medium_priority")}
        </Text>
        <Text style={styles.taxPrioritiesTaxText}>
          {i18n.t("high_priority")}
        </Text>
      </View>
      <SmallBoxInfo boxesInfos={fees} />
    </>
  );
}
