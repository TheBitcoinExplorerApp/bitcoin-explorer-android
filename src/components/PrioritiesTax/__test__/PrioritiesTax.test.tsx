/* eslint-disable import/no-extraneous-dependencies */
import renderer from "react-test-renderer";
import PrioritiesTax from "../PrioritiesTax";
import "expo-localization"; // Importa para garantir que o mock estÃ¡ sendo usado

describe("PrioritiesTax", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<PrioritiesTax />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
