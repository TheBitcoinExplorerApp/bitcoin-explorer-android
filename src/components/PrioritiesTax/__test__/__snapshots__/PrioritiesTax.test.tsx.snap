// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrioritiesTax renders correctly 1`] = `
[
  <View
    style={
      {
        "backgroundColor": "#1d2133",
        "borderRadius": 8,
        "flexDirection": "row",
        "justifyContent": "space-around",
        "padding": 16,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        {
          "color": "#FFF",
          "fontSize": 12,
        }
      }
    >
      Baixa Prioridade
    </Text>
    <Text
      style={
        {
          "color": "#FFF",
          "fontSize": 12,
        }
      }
    >
      MÃ©dia Prioridade
    </Text>
    <Text
      style={
        {
          "color": "#FFF",
          "fontSize": 12,
        }
      }
    >
      Alta Prioridade
    </Text>
  </View>,
  <View
    style={
      {
        "flexDirection": "row",
        "justifyContent": "space-around",
        "width": "100%",
      }
    }
  />,
]
`;
